### DISCUSSION TITLE
Bring API integration that does not require the Mealie instance to be publicly available

### DISCUSSION THREAD

**Discussion by felixschndr** [mealie-recipes/mealie#3627] [Show and tell]
[Mealie](https://github.com/mealie-recipes/mealie) startet to support sending a recipe to a Bring shopping list with this [PR](https://github.com/mealie-recipes/mealie/pull/3448). However, this requires the Mealie instance to be publicly available (from the internet). Since many users want their self-hosted services to not be available from the internet, I chose to create an integration that works local only without the need to expose anything to the public.

It is available [here](https://github.com/felixschndr/mealie-bring-api). You can find the source code and a container image.

I am currently running this container image and like how it works.

*In my opinion* this is pretty great and I'd like more users to know about this and be able to use it. I was wondering if this would make a good addition to

- Mealie itself (as in integrate it into Mealie, both projects are written in Python) or
- Add my repository to the repositories of the [Mealie github organization](https://github.com/mealie-recipes) and/or
- Add a link from the [docs of Mealie](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions) to my integration

What do you think about this?

**Comment by MuellerNicolas** [comment]
Hi @felixschndr, I've tried out your Bring-API-Connector and I really like it.
I am not familiar with the Mealie project codebase. As I understand it, the connector should be integrated into the project itself. The reason for this is the duplication of the "add to shopping cart" feature.
![image](https://github.com/user-attachments/assets/e7d1a7d5-23ba-4aa0-859c-da29d05b8e6d)
It would be nice if the original "add to shopping cart" button could be replaced by the bring integration if the user uses the Bring API integration.
But the solution would probably take the most time, due to coordination with the Mealie developers and necessary code changes

**Reply by felixschndr** [reply]
Hey Nicolas,
I am glad you like the integration.
> As I understand it, the connector should be integrated into the project itself. 

Yeah, in an ideal situation my component would be directly integrated into Mealie and thus redundant.
I guess there are three options:
1. Leave it like it is now.
2. Do as stated in my original post
   > In my opinion this is pretty great and I'd like more users to know about this and be able to use it. I was wondering if this would make a good addition to
    > Mealie itself (as in integrate it into Mealie, both projects are written in Python) or
Add my repository to the repositories of the [Mealie github organization](https://github.com/mealie-recipes) and/or
Add a link from the [docs of Mealie](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions) to my integration 
3. Integrate the component into the Mealie codebase itself. This would be the cleanest and best solution but requires the most work obviously.

Sadly, I currently don't have the resources to do 3. myself. I'd be more than happy to see someone tackle this. However, in the meantime I'd really enjoy 2. to be happening as at least some people like the integration as seen from the github reactions on my post. Friendly ping @michael-genson @boc-the-git @Kuchenpirat

**Reply by michael-genson** [reply]
We probably wouldn't want to integrate this into Mealie directly, I think this is best as an integration/plug-in. Aside from its recipe/data management capabilities, Mealie is more geared towards enabling workflows rather than directly implementing them. It would be different if there were some shopping API standard, but AFAIK that doesn't exist.

Adding it to the community docs I think is a great idea!

**Reply by felixschndr** [reply]
> Adding it to the community docs I think is a great idea!

Can you specify which docs exactly you mean by that? [Those docs](https://docs.mealie.io/documentation/getting-started/introduction/)?

-----
I guess I would add it somewhere in [here](https://docs.mealie.io/documentation/getting-started/features/#integrations) which a ref to that [here](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions) (where it states about Bring). What do you think about this @michael-genson?
Can you please show me where the source code for the docs lie? Then I can make a PR there

**Reply by michael-genson** [reply]
We have a community guides section in the docs, e.g. https://docs.mealie.io/documentation/community-guide/ios/

In the repo: https://github.com/mealie-recipes/mealie/tree/mealie-next/docs/docs/documentation/community-guide

**Reply by felixschndr** [reply]
Okay, I will open a PR to add my integration to that in the near future

**Reply by felixschndr** [reply]
See https://github.com/mealie-recipes/mealie/pull/4920
